include/rocksdb/metadata.h

```diff
@@ -16,24 +16,105 @@
 
 namespace ROCKSDB_NAMESPACE {
 
+// Basic identifiers and metadata for a file in a DB. This only includes
+// information considered relevant for taking backups, checkpoints, or other
+// services relating to DB file storage.
+struct FileStorageInfo {
+  // The name of the file within its directory (e.g. "123456.sst")
+  std::string relative_filename;
+  // The directory containing the file, without a trailing '/'. This could be
+  // a DB path, wal_dir, etc.
+  std::string directory;
+
+  // The id of the file within a single DB. Set to 0 if the file does not have
+  // a number (e.g. CURRENT)
+  uint64_t file_number = 0;
+  // The type of the file as part of a DB.
+  FileType file_type = kTempFile;
+
+  // File size in bytes. See also `trim_to_size`.
+  uint64_t size = 0;
+  // If true, the file on disk is allowed to be larger than `size` but only
+  // the first `size` bytes should be used for the current context. If false,
+  // the file is corrupt if size on disk does not equal `size`.
+  //
+  // If a CURRENT file needs to be regenerated to have a functioning DB, its
+  // size=0 and trim_to_size=true.
+  bool trim_to_size = false;
+
+  // This feature is experimental and subject to change.
+  Temperature temperature = Temperature::kUnknown;
+
+  // The checksum of a SST file, the value is decided by the file content and
+  // the checksum algorithm used for this SST file. The checksum function is
+  // identified by the file_checksum_func_name. If the checksum function is
+  // not specified, file_checksum is "0" by default.
+  std::string file_checksum;
+
+  // The name of the checksum function used to generate the file checksum
+  // value. If file checksum is not enabled (e.g., sst_file_checksum_func is
+  // null), file_checksum_func_name is UnknownFileChecksumFuncName, which is
+  // "Unknown".
+  std::string file_checksum_func_name;
+};
+
+namespace detail {
+struct SstFileMetaDataCompat : public FileStorageInfo {
+  // DEPRECATED: The name of the file within its directory with a
+  // leading slash (e.g. "/123456.sst"). Use relative_filename instead.
+  std::string name;
+
+  // DEPRECATED: replaced by `directory`
+  std::string& db_path = directory;
+
+  SstFileMetaDataCompat() {}
+
+  SstFileMetaDataCompat(const std::string& _file_name,
+                        const std::string& _directory) {
+    if (!_file_name.empty()) {
+      if (_file_name[0] == '/') {
+        name = _file_name;
```

Comment:
According to CircleCI, my tests pass on Windows. I didn't create the convention of calling a "file name" a final path segment with a leading '/', but I'm trying to get rid of that, because it's so unusual and potentially problematic. (If you intend `rm /dir/foo`, then `rm foo` is safer than `rm /foo`.)

Reply:
See also DB::GetLiveFiles